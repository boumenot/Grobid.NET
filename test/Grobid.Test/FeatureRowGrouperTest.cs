using ApprovalTests;
using ApprovalTests.Reporters;
using Newtonsoft.Json;
using Xunit;

using Grobid.NET;

namespace Grobid.Test
{
    [UseReporter(typeof(BeyondCompare4Reporter))]
    public class FeatureRowGrouperTest
    {
        [Fact]
        public void Simple()
        {
            var testSubject = new FeatureRowGrouper();
            var featureRows = new[]
            {
                new FeatureRow { Classification = "title", Value = "title" },
                new FeatureRow { Classification = "email", Value = "email" },
            };

            var groups = testSubject.Group(featureRows);
            Approvals.VerifyJson(JsonConvert.SerializeObject(groups));
        }

        [Fact]
        public void ConsecutiveClassifications()
        {
            var testSubject = new FeatureRowGrouper();
            var featureRows = new[]
            {
                new FeatureRow { IsStart = true, Classification = "email", Value = "email1" },
                new FeatureRow { IsStart = true, Classification = "email", Value = "email2" },
            };

            var groups = testSubject.Group(featureRows);
            Approvals.VerifyJson(JsonConvert.SerializeObject(groups));
        }

        [Fact]
        public void DifferentClassifications()
        {
            var testSubject = new FeatureRowGrouper();
            var featureRows = new[]
            {
                new FeatureRow { IsStart = true, Classification = "email", Value = "email1" },
                new FeatureRow { Classification = "email", Value = "email1" },
                new FeatureRow { Classification = "email", Value = "email1" },
                new FeatureRow { Classification = "email", Value = "email1" },
                new FeatureRow { IsStart = true, Classification = "title", Value = "title1" },
                new FeatureRow { Classification = "title", Value = "title1" },
                new FeatureRow { Classification = "title", Value = "title1" },
            };

            var groups = testSubject.Group(featureRows);
            Approvals.VerifyJson(JsonConvert.SerializeObject(groups));
        }

        [Fact]
        public void DifferentClassificationsWithoutStart()
        {
            var testSubject = new FeatureRowGrouper();
            var featureRows = new[]
            {
                new FeatureRow { Classification = "email", Value = "email1" },
                new FeatureRow { Classification = "email", Value = "email1" },
                new FeatureRow { Classification = "email", Value = "email1" },
                new FeatureRow { Classification = "email", Value = "email1" },
                new FeatureRow { Classification = "title", Value = "title1" },
                new FeatureRow { Classification = "title", Value = "title1" },
                new FeatureRow { Classification = "title", Value = "title1" },
            };

            var groups = testSubject.Group(featureRows);
            Approvals.VerifyJson(JsonConvert.SerializeObject(groups));
        }
    }
}
