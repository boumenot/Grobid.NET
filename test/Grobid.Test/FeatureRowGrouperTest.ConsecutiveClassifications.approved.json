[
    [
        {
            "Value": "email1",
            "Classification": "email",
            "IsStart": true
        }
    ],
    [
        {
            "Value": "email2",
            "Classification": "email",
            "IsStart": true
        }
    ]
]